### ISSUE TITLE
 "You are not going to set up a public gateway. However, your web UI is open to the whole internet."

### ISSUE THREAD

**Issue by kopy-kate** [HelloZeroNet/ZeroNet#1073]
** You are not going to set up a public gateway. However, <b>your Web UI is open to the whole Internet.</b> Please check your configuration.**

I'm getting this error message ever since I upgraded to ZeroNet 0.5.7.

That means I can't access the default gateway 127.0.0.1:43110.. I can only access it through my ZeroNet proxy.. http://www.kittyseedbox.tk:43110/

Is that a new security feature? What's wrong with my configuration, anyway?

**Comment by HelloZeroNet** [comment]
For public proxies it's recommended to enable the multiuser plugin (just rename plugins/disabled-Multiuser to plugins/Multiuser)
If you get invalid host error start it using `--ui_host 127.0.0.1 www.kittyseedbox.tk`

**Comment by kopy-kate** [comment]
hmm.. I'm still getting:

Forbidden
Invalid host: 127.0.0.1:43110

I've renamed the multiuser folder (I had done it previously but I guess the new update renamed it to disabled-?) and I'm running Sudo zeronet.py --ui_host 127.0.0.1 www.kittyseedbox.tk   .. as you suggested. 

None of the gateways are accessible now.. neither the localhost nor the proxy domain. :(

**Comment by kopy-kate** [comment]
I should add - I've tried endless combinations, but so far the only ones that seem to work are:

**--ui_ip '*'** (which now only allows me to access www.kittyseedbox.tk, but not 127.0.0.1)

and

**--ui_ip 0.0.0.0** (which allows me to access 127.0.0.1, but not www.kittyseedbox.tk)

**--ui_host** does absolutely nothing, and all gateways are forbidden no matter what you type in it.

I used to be able to access both in the previous version.. it was just handy I guess, as I could refresh and view changes I made to my local web files without having to publish them.

**Comment by HelloZeroNet** [comment]
Yes, sorry `--ui_host 127.0.0.1:43110 www.kittyseedbox.tk:43110`

**Comment by kopy-kate** [comment]
It's strange. --ui_host 127.0.0.1:43110 works fine, but not www.kittyseedbox.tk:43110

in other words:
**--ui_host 127.0.0.1:43110 www.kittyseedbox.tk:43110**    will only allow the localhost to be viewed, but not the domain proxy.

--ui_host www.kittyseedbox.tk:43110 will not allowed either of them to be viewed.

it seems to have a distaste for anything other than IP numbers.

**Comment by kopy-kate** [comment]
Think I figured it out. The only way to get it to work:
**sudo python zeronet.py --ui_host 127.0.0.1:43110 www.kittyseedbox.tk _--ui_ip '*'_**

**Comment by grez911** [comment]
@kopy-kate @HelloZeroNet So the issue has been solved?

**Comment by gitbugged** [comment]
@kopy-kate you should not have to run Python as root (sudo), it's bad for security.

**Comment by 6543** [comment]
@HelloZeroNet add a general "HowTo setup a Public Proxy" to the REDAME.md ?

**Comment by BenMcLean** [comment]
If ui_host is not specified, then I can connect over LAN, but connections from online get

> Forbidden
> Invalid host: mysubdomain.example.com:myport

But if any string I've tried is specified for ui_host then all connections get that message for whatever their hostname or IP happens to be.

I think an error message with more of an explanation of exactly why the hostname is invalid might help.

**Comment by imachug** [comment]
@HelloZeroNet Can this be closed?

**Comment by imachug** [comment]
@HelloZeroNet What about closing this?

**Comment by carbolymer** [comment]
Still does not work. I've enabled `Multiuser` plugin and added host to `/etc/zeronet.conf`:
```
[global]
fileserver_ip_type = ipv4
tor = disable
log_level = INFO
language = pl
fileserver_port = 38464
data_dir = /var/lib/zeronet
log_dir = /var/log/zeronet

ui_ip = *
ui_host =
  10.0.0.2
  myserver.local:43110
# ui_password = anypassword

tor_controller = 127.0.0.1:9051
tor_proxy = 127.0.0.1:9050
```
and getting in logs:
```
Ui.UiServer Error 403: Invalid host: myserver.local:43110
````

**Comment by icf20** [comment]
i am getting this error while running zeronet on docker

all setting default

**Comment by Jerrk** [comment]
i also get this message with docker

**Comment by chiva** [comment]
This is "fine and expected" with Docker, as you normally want the service to bind to all available interfaces
BUT make sure uPnP is disabled in your router, as ZeroNet tries to punch a hole to allow incoming connections, so you want to avoid that, as it can then really expose it to the open internet.
